kiosk/kiosk_browser/connection.py

```diff
@@ -0,0 +1,44 @@
+import requests
+import tempfile
+import threading
+import time
+
+check_connection_url = 'http://captive.apple.com/'
+sleep_time = 5
+
+class Connection():
+
+    def __init__(self, set_captive_portal_url):
+        self._is_connected = False
+        self._set_captive_portal_url = set_captive_portal_url
+
+    def start_daemon(self):
+        thread = threading.Thread(target=self._check, args=[])
+        thread.daemon = True
+        thread.start()
+
+    def is_connected(self):
+        return self._is_connected
+
+    def _check(self):
+        # Initial sleep to be sure _set_captive_portal_url is accessible
+        time.sleep(1)
+
+        while True:
+            try:
+                r = requests.get(check_connection_url, allow_redirects = False)
+
+                if r.status_code == 200:
+                    self._is_connected = True
+                    self._set_captive_portal_url('')
+                elif r.status_code in [301, 302, 303, 307]:
+                    self._is_connected = False
+                    self._set_captive_portal_url(r.headers['Location'])
+                else:
```

Comment:
I wonder if an additional case for `511 Network Authentication Required` is needed: https://tools.ietf.org/html/rfc6585#section-6

In the case, I guess the captive portal URL would need to be the test URL itself.

Reply:
I could not find good data on whether this is used much in practice. Firefox does not seem to support it, based on this issue: https://bugzilla.mozilla.org/show_bug.cgi?id=728658

Reply:
Do you know if we could get information concerning captive portals actually used ? If every captive portal already used gives a redirect code and a location, or if some of them gives back a 511.